openml/study/study.py

```diff
@@ -1,19 +1,35 @@
+import collections
+import openml
+import xmltodict
 
-class OpenMLStudy(object):
-    '''
-    An OpenMLStudy represents the OpenML concept of a study. It contains
-    the following information: name, id, description, creation date,
-    creator id and a set of tags.
-
-    According to this list of tags, the study object receives a list of
-    OpenML object ids (datasets, flows, tasks and setups).
 
-    Can be used to obtain all relevant information from a study at once.
+class OpenMLStudy(object):
 
-    Parameters
-       ----------
-        id : int
+    def __init__(self, study_id, alias, main_knowledge_type, benchmark_suite, 
+                 name, description, creation_date, creator, tags, data, tasks, 
+                 flows, setups, runs):
+        """
+        An OpenMLStudy represents the OpenML concept of a study. It contains
+        the following information: name, id, description, creation date,
+        creator id and a set of tags.
+    
+        According to this list of tags, the study object receives a list of
+        OpenML object ids (datasets, flows, tasks and setups).
+    
+        Can be used to obtain all relevant information from a study at once.
+    
+        Parameters
+        ----------
+        study_id : int
             the study id
+        alias : str (optional)
+            a string ID, unique on server (url-friendly)
+        main_knowledge_type : str
+            the knowledge type (e.g., task, run) that is core in this study.
```

Comment:
Is this only task or run? If yes, could you please explicitly state this? Also, I'm wondering whether this should be renamed to for example `main_entity_type` to highlight that this runs on different types of entities (which is in my opinion easier to relate to the different concepts like dataset, task etc. than knowledge).

Reply:
I think that the terms `entities` and `knowledge type` are not consistently documented within OpenML. Maybe the term `entity type` and `entities` would be better though

Reply:
Done. Please check the PR again, as I replaced the word knowledge type with entity type throughout the relevant files